### PR TITLE
Make Common version requirement less strict

### PR DESCRIPTION
**Pull request by SiarheiFedartsou**
https://github.com/mapbox/mapbox-navigation-native/issues/2786

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -18,7 +18,7 @@ let package = Package(
         )
     ],
     dependencies: [
-        .package(url: "https://github.com/mapbox/mapbox-common-ios.git", .exact(mapboxCommonVersion)),
+        .package(url: "https://github.com/mapbox/mapbox-common-ios.git", from: mapboxCommonVersion),
```
Comment (SiarheiFedartsou) [inline Package.swift:21]: According to SPM docs:
```
/// The following example allows the Swift Package Manager to select a version
/// like a  `1.2.3`, `1.2.4`, or `1.3.0`, but not `2.0.0`.
///
///    .package(url: "https://example.com/example-package.git", from: "1.2.3"),
```

Is it okay to allow minor versions in our cases(11.1.0 in this particular case can be installed)?

File: Package.swift [hunk 2]
```diff
@@ -5,9 +5,15 @@ import Foundation
...
 let version = "48.0.3"
-let mapboxCommonVersion = Version("11.0.1") 
+let mapboxCommonVersion = Version("11.0.1")
 let checksum = "9233bcb28551ded20157c22e78d5bdc0ff859e1e6423bed34461b3310f23cc22"
 
+func mapboxCommonVersionUpperBound(_ mapboxCommonVersion: Version) {
+    var version = mapboxCommonVersion
+    version.patch = 99999
+    return version
+}
```
Comment (dersim-davaod) [inline Package.swift:15]: Can we have more generic function? something similar to:

```suggestion
func versionRangeThatAllowsOnlyPatches(for version: Version) -> Range<Version> {
    version..<Version("\(version.major).\(version.minor + 1).0")
}
```
 
Then we can get rid of comment below and use it as:
 ```swift
 .package(url: "https://github.com/mapbox/mapbox-common-ios.git", versionRangeThatAllowsOnlyPatches(for: mapboxCommonVersion)),
```

**Review by dersim-davaod — CHANGES_REQUESTED** [review]

**Comment by dersim-davaod** [comment]
Additional comment regarding to the base branch. Currently it is set to `main`, but seems the changes should be landed into the `release-v2.0`. @1ec5 please correct me if I'm wrong.

File: Package.swift [hunk 3]
```diff
@@ -18,7 +22,7 @@ let package = Package(
         )
     ],
     dependencies: [
-        .package(url: "https://github.com/mapbox/mapbox-common-ios.git", .exact(mapboxCommonVersion)),
+        .package(url: "https://github.com/mapbox/mapbox-common-ios.git", versionRangeThatAllowsOnlyPatches(for: mapboxCommonVersion)),
```
Comment (1ec5) [inline Package.swift:25]: This is equivalent to `Package.Dependency.Requirement.upToNextMinor(from:)`, which would be clearer than the custom function used here. However, CocoaPods and Carthage don’t quite support the same semantics, at least not for post-v1.0 packages. The only common denominator that works regardless of package manager is `.upToNextMajor(from:)`, which is incidentally the [officially recommended](https://developer.apple.com/documentation/swift_packages/package/dependency/3582850-package#discussion) practice on iOS.

**Comment by 1ec5** [comment]
> Additional comment regarding to the base branch. Currently it is set to `main`, but seems the changes should be landed into the `release-v2.0`.

This repository has no release-v2.0 branch because every new release updates main. This does lead to some back-and-forth as v3_x_._y_ releases continue to be published, but it isn‘t a huge deal because the only thing SPM cares about is Package.swift’s current contents as of a particular tag.

**Review by dersim-davaod — APPROVED** [review]
LGTM

**Pull request merged**